http-server/src/main/java/org/triplea/modules/chat/Chatters.java

```diff
@@ -97,4 +107,73 @@ public boolean disconnectPlayerSessions(final UserName userName, final String di
         });
     return !sessions.isEmpty();
   }
+
+  /**
+   * Checks if a given chatter is muted, if so returns the {@code Instant} when the mute expires
+   * otherwise returns an empty optional
+   */
+  public Optional<Instant> getPlayerMuteExpiration(final InetAddress inetAddress) {
+    return getPlayerMuteExpiration(inetAddress, Clock.systemUTC());
+  }
+
+  @VisibleForTesting
+  Optional<Instant> getPlayerMuteExpiration(final InetAddress inetAddress, final Clock clock) {
+    return Optional.ofNullable(playerMutes.get(inetAddress))
+        .map(
+            muteInstant -> {
+              if (muteInstant.isAfter(clock.instant())) {
+                return muteInstant;
+              } else {
+                playerMutes.remove(inetAddress);
+                return null;
+              }
+            });
```

Comment:
This sounds like a good candidate for [Map#computeIfPresent](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html#computeIfPresent(K,java.util.function.BiFunction)):
```java
return Optional.ofNullable(playerMutes.computeIfPresent(inetAddress, (key, muteInstant) -> {
  if (muteInstant.isAfter(clock.instant())) {
    return muteInstant;
  }
  return null;
}));
```
Note that I haven't actually tested this code, but it should have the exact same behaviour if the javadocs are accurate

Reply:
Does that remove expired mutes? The final return null seems like we might be setting a null value, it would be better though to remove the entire entry.

Reply:
According to the Javadoc I linked it does.

Reply:
Cool, I did not know that. We can also use a ternary here as well to simplify further. 

Updated: 53b4f70e2

Reply:
I didn't know that until yesterday either, but I finally found a use-case for this method ^^